### DISCUSSION TITLE
Add number format support

### DISCUSSION THREAD

**Discussion by renakdup** [jeffreyvr/wp-settings#19] [Ideas]
Hi, thanks for this library. I found that your package doesn't support number format. I'm going to make PR with this.

**Comment by jeffreyvr** [comment]
Hi! You can use the text option and then set the type to number. Example: 

```php
$section->add_option('text', [
    'name' => 'option_1',
    'label' => __('Option 1', 'textdomain').
    'type' => 'number'
]);
```

**Reply by vikasbhvsr** [reply]
Hey @jeffreyvr ,

My use case is slightly different. I want to use `'type' => 'button'` for making ajax calls. Unfortunately. There's no way to pass `value` to the input. Am I missing something?

**Reply by jeffreyvr** [reply]
I your case, you might consider adding a custom option type. 
https://github.com/jeffreyvr/wp-settings?tab=readme-ov-file#adding-a-custom-option-type

But you can also pass a default value for an option by passing the `default` attribute.

**Reply by vikasbhvsr** [reply]
Yup, that's what I ended up doing. Thanks again.
